src/classes/ADDR_Seasonal_UTIL.cls

```diff
@@ -200,7 +200,8 @@ public with sharing class ADDR_Seasonal_UTIL {
             for (Address__c addr2 : listAddrS) {
                 if (addr == addr2)
                     continue;
-                if (addr.Parent_Account__c != addr2.Parent_Account__c)
+                if ((addr.Parent_Account__c != null && addr.Parent_Account__c != addr2.Parent_Account__c) || 
+                    (addr.Parent_Contact__c != null && addr.Parent_Contact__c != addr2.Parent_Contact__c))
```

Comment:
This is just a optimization to avoid comparing addresses in a bulk process that have different parent Contacts.